addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -53,191 +58,299 @@
-                                    getConfig().get(COMMAND), command.toString());
-                            scheduler.schedule(periodicExecutionRunnable, 0, TimeUnit.SECONDS);
+                                    config.getCommand(), command.toString());
+                            scheduler.schedule(new ExecutionRunnable(currentInput), 0, TimeUnit.SECONDS);
                         }
                     }
                 }
             }
         }
     }
 
-    @Override
-    public void initialize() {
+    class ExecStrLookup extends StrLookup {
 
-        if (executionJob == null || executionJob.isCancelled()) {
-            if (((BigDecimal) getConfig().get(INTERVAL)) != null && ((BigDecimal) getConfig().get(INTERVAL)).intValue() > 0) {
-                int polling_interval = ((BigDecimal) getConfig().get(INTERVAL)).intValue();
-                executionJob = scheduler.scheduleWithFixedDelay(periodicExecutionRunnable, 0, polling_interval,
-                        TimeUnit.SECONDS);
-            }
+        private String execInput;
+
+        public void setInput(String input) {
+            this.execInput = input;
         }
 
-        updateStatus(ThingStatus.ONLINE);
-    }
+        @Override
+        public String lookup(String key) {
+            String transform = null;
+            String format = null;
+
+            String parts[] = key.split(":");
+            String subkey = parts[0];
+
+            if (parts.length == 2) {
+                format = parts[1];
+            } else if (parts.length == 3) {
+                transform = parts[1];
+                format = parts[2];
+            }
 
-    @Override
-    public void dispose() {
-        if (executionJob != null && !executionJob.isCancelled()) {
-            executionJob.cancel(true);
-            executionJob = null;
+            try {
+                if ("exec-time".equals(subkey)) {
+                    // Transform is not relevant here, as our source is a Date
+                    if (format != null) {
+                        return String.format(format, Calendar.getInstance().getTime());
+                    } else {
+                        return null;
+                    }
+                } else if ("exec-input".equals(subkey)) {
+                    if (execInput != null) {
+                        String transformedInput = execInput;
+                        if (transform != null) {
+                            transformedInput = transformString(transformedInput, transform);
+                        }
+                        if (format != null) {
+                            return String.format(format, transformedInput);
+                        } else {
+                            return execInput;
+                        }
+                    } else {
+                        return null;
+                    }
+                } else {
+                    String transformedInput = itemRegistry.getItem(subkey).getState().toString();
```

Comment:
Do you think the Exec binding should do these transformations itself using item state from the item registry @kaikreuzer?

Reply:
> Do you think the Exec binding should do these transformations itself using item state from the item registry @kaikreuzer?

@kaikreuzer any idea's? Or shall we first get rid of #1637